docs/source/troubleshooting/index.rst

```diff
@@ -183,3 +183,68 @@ compare-and-set operation here.
          True
 
 Clients should be able to take the backup lock again after this step.
+
+Migrating from conservatively swept tables to thoroughly swept tables
+=====================================================================
+
+Background and Motivation
+-------------------------
+
+Long running read transactions need to be able to distinguish between cells that were never written to, and cells that
+have been overwritten and are now being cleaned by sweep. A solution to this is to leave a sentinel that will always be
+read no matter what timestamp the transaction is running at when deleting a cell along with the normal Atlas-level
+delete (empty byte-array). This is how tables running with conservative sweep work. The downside is that for data that
+is transient - i.e. gets written to and deleted fairly quickly - this metadata stays behind forever, and even if the
+table is empty from an Atlas level, it will still have all of this metadata in the underlying key value service,
+resulting in unbounded disk usage growth the more cells are written to and deleted from. It also means that more data
+is potentially loaded from the underlying KVS when doing range scans making queries worse as time goes on.
+
+An alternative solution is to take out locks on the immutable timestamp, such that sweep cannot progress further than
+any in progress transaction. This means we needn't write any of the sentinels for long running read transactions to
+find. This is how sweep interacts with tables that are marked with the "thorough" sweep strategy.
+
```

Comment:
In general for this section: I think this should not be described as an alternative approach, as the immutable timestamp is actually checked in both cases. 

Conservative Sweep _additionally_ wants to support read transactions, and if we do this we need to make sure sweep does not get rid of values that may be read for long running read transactions, hence the need for the unreadable timestamp (which you describe in the first paragraph). Actually conservative sweep uses for its sweep timestamp the minimum of the immutable and unreadable timestamps.